formatter_test.go

```diff
@@ -587,6 +587,120 @@ func TestFormatter_FloatFormat(t *testing.T) {
 	}
 }
 
+func TestFormatter_DigitSeparator(t *testing.T) {
+	type testCase struct {
+		name      string
+		separator DigitSeparator
+		format    FloatFormat
+		value     interface{}
+		wantText  string
+	}
+
+	testCases := []testCase{
+		{
+			name:      "float no decimal",
+			separator: DigitSeparatorUnderscore,
+			format:    FloatFormatDecimal,
+			value:     float32(12345678),
+			wantText:  "12_345_678",
+		},
+		{
+			name:      "float64 auto small exponent",
+			separator: DigitSeparatorUnderscore,
+			format:    FloatFormatAuto,
+			value:     float64(1.23456789),
+			wantText:  "1.234_567_89",
+		},
+		{
+			name:      "nofrac scientific",
+			separator: DigitSeparatorUnderscore,
+			format:    FloatFormatScientific,
+			value:     float32(12345678),
+			wantText:  "1.234_567_8e+07",
+		},
+		{
+			name:      "integer",
+			separator: DigitSeparatorUnderscore,
+			format:    FloatFormatAuto,
+			value:     int(12345678),
+			wantText:  "12_345_678",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			formatter := DefaultFormatter{
+				DigitSeparator: tc.separator,
+				FloatFormat:    tc.format,
+			}
+			formatData := formatter.buildFormatData(
+				&AssertionContext{},
+				&AssertionFailure{
+					Type:   AssertValid,
+					Actual: &AssertionValue{tc.value},
+				})
+			assert.Equal(t, tc.wantText, formatData.Actual)
+		})
+	}
+}
+
+func TestFormatter_DigitSeparatorType(t *testing.T) {
```

Comment:
We also need a separate test that check how separator is applied to different fliatand integer types (float32, float64, int, uint)